client/my-sites/hosting/sftp-card/index.js

```diff
@@ -51,13 +51,35 @@ export const SftpCard = ( {
 	// State for clipboard copy button for both username and password data
 	const [ isLoading, setIsLoading ] = useState( false );
 	const [ isPasswordLoading, setPasswordLoading ] = useState( false );
+	const [ isCopied, setIsCopied ] = useState( {
+		password: false,
+		url: false,
+		port: false,
+		username: false,
+	} );
 
 	const onDestroy = () => {
 		if ( password ) {
 			removePasswordFromState( siteId, currentUserId, username );
 		}
 	};
 
+	const onCopyPassword = () => {
+		setIsCopied( { password: true, url: false, port: false, username: false } );
+	};
+
+	const onCopyUrl = () => {
+		setIsCopied( { password: false, url: true, port: false, username: false } );
+	};
+
+	const onCopyPort = () => {
+		setIsCopied( { password: false, url: false, port: true, username: false } );
+	};
+
+	const onCopyUsername = () => {
+		setIsCopied( { password: false, url: false, port: false, username: true } );
+	};
```

Comment:
Nitpick: I generally don't like early abstractions, but since we're already slightly copying the same code 4 times, it might be a good indicator that an abstraction here could help 🙂 

Something like this can work:

```js
const onCopy = ( field ) => setIsCopied( { ...isCopied, [ field ]: true } );
```

Then each `ClipboardButton` component should call to that function and provide the name of the affected field. Example:
```jsx
<ClipboardButton
	className="sftp-card__copy-button"
	text={ password }
	onCopy={ () => onCopy( 'password' ) }
	compact
>[...]</ClipboardButton>
```

Good thing about this approach is that the "Copied" label will be kept after copying a different field.